### DISCUSSION TITLE
How to implement infinite scrolling?

### DISCUSSION THREAD

**Discussion by fablb** [ghiscoding/Angular-Slickgrid#1198] [Q&A] [answered]
Hi,

I've seen many examples in the wiki to do lazy loading using pagination with for example the GraphQL backend service. Nevertheless I didn't saw any example to do lazy loading with infinite scrolling.

At first I thought to make my own custom backend service using the interface provided and then use the onViewportChanged event from slickgrid, but the interface seems to be made by design for pagination.

I also saw the remotemodel from slickgrid but I'm not sure if it's still suited.

All the operations like filtering, sorting, etc. would be done server side.

Any suggestions would be greatly appreciated.

**Comment by ghiscoding** [comment]
You didn't see any example or docs about infinite scrolling simply because it's not a feature of SlickGrid and is not supported. 

The remotemodel is simply a custom DataView and is not recommended to use at all (I never used it), it was only provided because someone used it from SlickGrid in the past and wanted it in Angular-Slickgrid.

This question was also asked in the past and the answer was the same #156 

You are correct in saying that it uses lazy loading with pagination for backend services, the way I've implemented that was simply to change data by calling the DataView `setItems` and whenever we change page then we replace the entire data with `setItems`. You could try to do that with the DataView when scrolling with the `onScroll` event but please be aware that the DataView does not accept duplicate Ids so you would not be able to have the same data twice (i.e. on top and bottom)

Good luck

**Reply by fablb** [reply]
Are there any differences in the rendering process when we set data to the default variable (like `dataset: any[]`) vs when we use `setItems` from DataView?

Thanks.

**Reply by ghiscoding** [reply]
it's roughly the same, it does extra logic and yes you should prefer the `dataset` as much as possible so that Angular-Slickgrid stays in sync

**Comment by ghiscoding** [comment, marked as answer]
Infinite Scroll is now supported for both approach of either a local JSON dataset or with Backend Services. See new [v8.5.0](https://github.com/ghiscoding/Angular-Slickgrid/releases/tag/v8.5.0) release for more info
